File: modules/linux/hello.rs
// SPDX-License-Identifier: GPL-2.0

/// Linux lkm module
use kernel::prelude::*;

module! {
    type: Hello,
    name: "hello",
    author: "Schubert Anselme <schubert@anselm.es>",
    description: "Linux lkm module",
    license: "GPL",
}

struct Hello;

impl kernel::Module for Hello {
    fn init(_module: &'static ThisModule) -> Result<Self> {
        pr_info!("lkm - hello world\n");
        Ok(Self)
    }
}

impl Drop for Hello {
    fn drop(&mut self) {
        pr_info!("lkm - goodbye world\n");
    }
}

File: pkg/client/web/Sources/hello.swift
// SPDX-License-Identifier: GPL-3.0
//
//  hello.swift
//  helloweb
//
//  Created by Schubert Anselme on 2024-11-11.
//

import Foundation
import Ignite

struct Hello: Site {
  var name = "hello web"
  var titleSuffix = " - hello"
  var url = URL(string: ProcessInfo.processInfo.environment["URL"] ?? "http://localhost:8080")!
  var author = "Schubert Anselme"

  // note: configuration
  var builtInIconsEnabled = true
  var syntaxHighlighters = [SyntaxHighlighter.swift, .python, .ruby]
  var feedConfiguration = FeedConfiguration(
    mode: .full, contentCount: 20, image: .init(url: "/images/icon32.png", width: 32, height: 32))
  var robotsConfiguration = Robots()

  // note: homepage
  var homePage = Home()

  // todo: pages

  // note: theme & layout
  var theme = DefaultTheme()
  var layouts: [any ContentPage] { DefaultLayout() }
}
